src/client/components/forms/type-editor/identifier-type.tsx

```diff
@@ -0,0 +1,441 @@
+	const handleAddParent = useCallback(() => {
+		setShowModal(true);
+	}, []);
+
+	// Callback function for closing the modal, the state of the modal should alse be reset
+	const handleModalClose = useCallback(() => {
+		setSelectedParentType(null);
+		setChildOrder(0);
+		setShowModal(false);
+	}, []);
+
+	// Function to handle parent type selection in ParentType modal
+	const handleParentTypeChange = useCallback((selectedOption) => {
+		if (selectedOption) {
+			setSelectedParentType(selectedOption.id);
+		}
+		else {
+			setSelectedParentType(null);
+		}
+	}, [selectedParentType]);
+
+	// Function to handle child order input in ParentType modal
+	const handleChildOrderChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+		const value = parseInt(event.target.value, 10);
+		setChildOrder(isNaN(value) ? 0 : value);
+	}, [formData, childOrder]);
+
+	// Function to handle parent removal
+	const handleRemoveParent = useCallback(() => {
+		setFormData((prevFormData) => ({
+			...prevFormData,
+			childOrder: 0, parentId: null
+		}));
+		setChildOrder(0);
+		setSelectedParentType(null);
+	}, [formData]);
+
+	const handleEditParent = useCallback(() => {
+		setShowModal(true);
+	}, []);
+
+	// Function to handle parent type and child order edit submission
+	const handleModalSubmit = useCallback(() => {
+		if (selectedParentType !== null) {
+			setFormData((prevFormData) => ({
+				...prevFormData,
+				childOrder, parentId: selectedParentType
+			}));
+			setShowModal(false);
+		}
+	}, [formData, childOrder, selectedParentType]);
+
+	const handleInputChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+		  const {name, value} = event.target;
+		  setFormData((prevFormData) => ({
+			...prevFormData,
+			[name]: value
+		  }));
+	}, [formData]);
+
+	const handleDeprecatedChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+		const {value} = event.target;
+		setFormData((prevFormData) => ({
+			...prevFormData,
+			deprecated: value === 'true'
+		}));
+	}, [formData]);
+
+	const errorAlertClass = classNames('text-center', 'margin-top-1', {'d-none': !errorMsg.length});
+
+	const getEntityTypeLabel = useCallback(option => option.name, []);
+
+	const getEntityTypeValue = useCallback(option => option.name, []);
+
+	const getParentTypeValue = useCallback(option => option.id, []);
+
+	// Callback function to format the option label
+	const formatParentTypeOptionLabel = useCallback(option => (
+		<div className="small">
+			<div>
+				<strong>Label:&nbsp;</strong>{option.label}
+			</div>
+			<div>
+				<strong>Description:&nbsp;</strong>{option.description}
+			</div>
+		</div>
+	), []);
+
+	const handleEntityTypeChange = useCallback((selectedOption) => {
+		if (selectedOption) {
+			setFormData({...formData, entityType: selectedOption.name});
+		}
+		else {
+			setFormData({...formData, entityType: null});
+		}
+	}, [formData]);
+
+	function isValid() {
+		return Boolean(formData.entityType);
```

Comment:
At the outset, I wanted an `isValid` function to check those required fields on the form which cannot directly be checked by adding a `required` flag. In case of the relationship type editor, there were two such fields: a `sourceEntityType` and a `targetEntityType`. In this case, we only have one such field: `entityType`.

I agree the name `isValid` itself is a bit more general(and maybe misleading), perhaps it should have been something like `checkRequiredFields`. We can also get rid of this function here, but I went ahead with it just to make it similar to the relationship type editor code.

Reply:
Ideally it should check all the required fields, but the combination of the `required` attribute on the inputs and the error thrown by the server is probably sufficient.
Trying to prove that point, I edited the html to remove the required attribute on all the inputs (but selecting an entity type to satisfy `isValid`) and I can submit with empty values, and get an error back from the server:
<img width="746" alt="image" src="https://github.com/metabrainz/bookbrainz-site/assets/6179856/3c0f2073-5c7e-449d-a5f2-96d19b6627f5">


IMO this is sufficient for this type of interface, and if proper validation should be done it should be done on the server-side.

Reply:
so what I mean by required fields is this: for non-select fields, there is prompt which occurs like this in case we try to submit the form without entering anything in that field:
![Screenshot 2023-09-18 215839](https://github.com/metabrainz/bookbrainz-site/assets/56965261/862b0e14-cb58-4d43-8b0c-2b68933a22d9)

For the select type fields, there is no such validation. In order to prevent sending incomplete data to backend, we should probably use this function to do this in frontend itself:
![Screenshot 2023-09-18 at 22-02-12 BookBrainz – The Open Book Database](https://github.com/metabrainz/bookbrainz-site/assets/56965261/7a6fa73c-f792-4291-b951-e0752e1ea24e)

Reply:
Although you're right, ideally we should be doing for all the fields. The only reason I went for this is because for the non-select fields at least there is a prompt which tells us `Please fill out this field.`, while for the select field, there is no such field.

Reply:
> so what I mean by required fields is this: for non-select fields, there is prompt which occurs like this in case we try to submit the form without entering anything in that field:

Yep, completely understand, tested and approved :)
I just tried to break it by manually modifying the HTML in my page to allow empty inputs, which would not normally happen in any reasonable circumstance I can imagine.